### PR TITLE
Feature/778 refactor register client

### PR DESCRIPTION
**Pull request by sigridbra**
## Description

Add methods to get users by the lookup criterias from userService. 
The tests are plenty - and generated by copilot. 

## Related Issue(s)
- #778 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

## Summary by CodeRabbit

* **New Features**
  * Retrieve user/party info by UUID, user ID, username, SSN and perform batch lookups; results use a dedicated parties response and are filtered to user-related party types.

* **Chores**
  * Bumped Register.Contracts dependency to a newer version.

* **Tests**
  * Expanded unit tests for the new lookups and edge cases; added HTTP request scenarios for user retrieval.

### CODE REVIEW

File: test/Altinn.Profile.Tests/Profile.Integrations/Register/RegisterClientTests.cs [hunk 1]
```diff
@@ -632,5 +634,424 @@ public async Task GetOrganizationNumberByPartyUuid_WhenCancellationRequested_Thr
...
+            var result = await client.GetUserParty(userUuid, TestContext.Current.CancellationToken);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(expectedParty.PartyId, result.PartyId);
+            Assert.Equal(expectedParty.Uuid, result.Uuid);
+            Assert.Equal(expectedParty.Type, result.Type);
+            Assert.Equal(HttpMethod.Post, sentRequest.Method);
+            Assert.Contains("v2/internal/parties/query?fields=person,party,user,si", sentRequest.RequestUri.ToString());
+            Assert.True(sentRequest.Headers.Contains("PlatformAccessToken"));
```
Comment (hggutvik) [inline test/Altinn.Profile.Tests/Profile.Integrations/Register/RegisterClientTests.cs]: This alternative should improve the diagnostic value on test failure. Rather than `"Expected: True, Actual: False"`, it reports the actual headers present on failure. 
```suggestion
            Assert.Contains(sentRequest.Headers, h => h.Key == "PlatformAccessToken");
```
